firedrake/preconditioners/asm.py

```diff
@@ -66,6 +80,129 @@ def applyTranspose(self, pc, x, y):
         self.asmpc.applyTranspose(x, y)
 
 
+class ASMStarPC(ASMPatchPC):
+    '''Patch-based PC using Star of mesh entities implmented as an
+    :class:`ASMPatchPC`.
+
+    ASMStarPC is an additive Schwarz preconditioner where each patch
+    consists of all DoFs on the topological star of the mesh entity
+    specified by `pc_star_construct_dim`.
+    '''
+
+    _prefix = "pc_star_"
```

Comment:
I think the convention we have is that you don't need `pc_` here. So for example if you had `star_` then you'd end up with:

```
-mg_levels_pc_type python
-mg_levels_pc_python_type ASMStarPC
-mg_levels_star_pc_type asm # this is default
-mg_levels_star_sub_pc_type lu
-mg_levels_star_sub_pc_factor_shift_type nonzero
```

The python preconditioner has prefix `-mg_levels_` and the inner preconditioner then has prefix `-mg_levels_star_`. I think that reduces the amount of typing.

Reply:
Did that in #1830.